lib/animina_web/live/story_live.ex

```diff
@@ -284,6 +303,188 @@ defmodule AniminaWeb.StoryLive do
     end
   end
 
+  def handle_event("correct_errors", _params, socket) do
+    process_story(
+      socket,
+      "Correct any spelling, grammar, case, and punctuation errors in the story.",
+      socket.assigns.live_action
+    )
+  end
+
+  def handle_event("improve_funny", _params, socket) do
+    process_story(socket, "Improve the story to be funnier.", socket.assigns.live_action)
+  end
+
+  def handle_event("improve_exciting", _params, socket) do
+    process_story(socket, "Improve the story to be more exciting.", socket.assigns.live_action)
+  end
+
+  def handle_event("lengthen_story", _params, socket) do
+    process_story(socket, "Lengthen the story.", socket.assigns.live_action)
+  end
+
+  def handle_event("shorten_story", _params, socket) do
+    process_story(socket, "Shorten the story.", socket.assigns.live_action)
+  end
+
+  defp process_story(socket, prompt, :new) do
+    Process.send_after(self(), {:render_generating_story, 1}, 1000)
+    headline = Headline.by_id!(socket.assigns.form.params["headline_id"])
+
+    socket =
+      case ChatCompletion.request_stories(
+             headline.subject,
+             socket.assigns.form.params["content"],
+             prompt
+           ) do
+        {:ok, task} ->
+          updated_params =
+            Map.update!(socket.assigns.form.params, "content", fn _content ->
+              ""
+            end)
+
+          socket
+          |> assign(:current_request, task)
+          |> assign(:form, Form.validate(socket.assigns.form, updated_params))
+          |> assign(:generating_story, true)
+          |> assign(:show_buttons, false)
+
+        {:error, _} ->
+          socket
+      end
+
+    {:noreply, socket}
+  end
+
+  defp process_story(socket, prompt, :edit) do
+    Process.send_after(self(), {:render_generating_story, 1}, 1000)
+
+    socket =
+      case ChatCompletion.request_stories(
+             socket.assigns.story.headline.subject,
+             socket.assigns.story.content,
+             prompt
+           ) do
+        {:ok, task} ->
+          new_story =
+            if socket.assigns.story.photo == nil do
+              %{
+                "photo" => %{
+                  "_form_type" => "create",
+                  "_ignored" => "true",
+                  "_persistent_id" => "0",
+                  "user_id" => "1be087b1-f935-4f87-93e0-1129d8dcb56c"
```

Comment:
I didn't dive deep into the code but is this fixed `user_id` intentional?

Reply:
@wintermeyer That was a mistake from my end sorry. Let me fix it